apps/angular/src/app/features/anime/pages/dashboard/dashboard-page.component.ts

```diff
@@ -41,14 +87,124 @@ export class DashboardPageComponent {
 
 	public constructor() {
 		this.animes$ = this.createAnimeStream();
+		this.searchValue.set(this.queryParams.search);
+	}
+
+	/**
+	 * Get anime list.
+	 * @param params Query params.
+	 */
+	private getAnimeList(params: AnimeParams): Observable<Pagination<Anime>> {
+		return this.animeService.getAnimeList({
+			limit: params.limit,
+			offset: params.offset,
+			search: params.search,
+			typeIn: params.typeIn,
+			sort: params.sort,
+		});
 	}
 
 	private createAnimeStream(): Observable<Pagination<Anime>> {
-		this.isLoading$.next(true);
+		this.isLoading.set(true);
 
-		return this.animeService.getAnimeList().pipe(
-			takeUntilDestroyed(this.destroyRef),
-			finalize(() => this.isLoading$.next(false)),
+		return this.activeRoute.queryParams.pipe(
+			distinctUntilChanged(),
+			map(query => this.transformQueryParams(query)),
+			switchMap(params =>
+				this.getAnimeList(params).pipe(
+					finalize(() => {
+						this.isLoading.set(false);
+						this.selectedTypes.set(this.queryParams.typeIn?.split(',') as AnimeType[] ?? []);
```

Comment:
is it really necessary to try to maintain the logic of API DTO when we work with query?